AcademicReward/Logic/ShopLogic.cs

```diff
@@ -42,9 +42,11 @@ public LogicErrorType AddItem(object obj)
                 return LogicErrorType.InvalidLevel;
             }
             
-            toBeAdded = new ShopItem(AddItemVals[0], AddItemVals[1], cost, level, null);
-            
-            
+            toBeAdded = new ShopItem(AddItemVals[0], AddItemVals[1], cost, level, 
+                MauiProgram.Profile.GroupList.ElementAt( int.Parse(AddItemVals[5]) ) );
```

Comment:
You could always do a TryParse here, that way if anything were to go wrong, the whole application wouldn't crash. Not sure how it would go wrong, but you never know